In [6]:
from textblob import TextBlob

text = "Revenue increased but we are cautious about future growth"

blob = TextBlob(text)
print("Sentiment:", blob.sentiment)

Sentiment: Sentiment(polarity=0.0, subjectivity=0.125)


## Polarity = 0.0 (Neutral Sentiment) 
- Polarity measures the emotional weight of a sentence on a scale from -1.0 (highly negative) to 1.0 (highly positive). 
- A score of 0.0 indicates that the sentence is strictly neutral.

In your specific text, "Revenue increased" (positive) is balanced out by "cautious about future growth" (negative). TextBlob often averages these opposing sentiments, resulting in a net neutral score of zero.



## Subjectivity = 0.125 (Mostly Objective)

- Subjectivity measures how much of the text is based on personal opinion versus factual information, ranging from 0.0 (very objective/factual) to 1.0 (very subjective/opinionated)

# Textblob  not reliable

In [3]:
tweets = [
    "Stock is going to the moon.",
    "Market crash coming",
    "Not sure about this stock"
]

from textblob import TextBlob

# Gives wrong answer
for t in tweets:
    print(t, "->", TextBlob(t).sentiment.polarity)


Stock is going to the moon -> 0.0
Market crash coming -> 0.0
Not sure about this stock -> -0.25
